### DISCUSSION TITLE
Integration with Astro

### DISCUSSION THREAD

**Discussion by loxy** [codyhouse/codyhouse-framework#134] [General]
Hi!

Can you outline how you would use / integrate your components with [Astro](https://astro.build/)? Do you would transform your components to Vue or React before using? 

Best regards,
Kersten

**Comment by peiche** [comment]
Hi, I have a starter theme for Gatsby (which uses React under the hood) [here](https://github.com/peiche/gatsby-starter-codyhouse). I didn't find it necessary to rewrite any of the Codyhouse component scripts for integration. I'm not familiar with Astro, but feel free to take a look at my repo; maybe it'll give you some ideas. :)

Edit: they do also mention integration with Vue and React in the [docs](https://codyhouse.co/ds/docs/components#vue-react).

**Reply by loxy** [reply]
Thanks Paul. Took a quick look. But didn't found a example of integrating a Codyhouse component. I'm aware of that docs. That why I asked myself if transforming a Codyhouse component to React or Vue before usage in Astro is the right way. Or can I easily use it directly?

**Reply by peiche** [reply]
You're right, I think I had to remove the example component to comply with Codyhouse licensing rules. I believe you can use the components directly, it's only for the markup where you'll have to put the HTML inside a React component .jsx or .tsx file.

**Reply by loxy** [reply]
But you can use the normal licence for that, right? Or do you mean to expose the example at a public web page? 

And I cannot use the components directly with that IFEE everywhere in their code. But I'm still trying out...

**Reply by peiche** [reply]
It's against their license to include components in a general use theme, like for WordPress or Gatsby. If you're building a website for someone or yourself, you can use their components.
